packages/react/src/components/PaginationNav/PaginationNav.js

```diff
@@ -0,0 +1,341 @@
+  if (page < 2 || page > totalItems - 3) reservedSpots -= 1;
+
+  const shownItems = itemsThatFit - reservedSpots;
+  const hiddenItems = totalItems - shownItems - (itemsThatFit < 5 ? 1 : 2);
+
+  let frontHidden = Math.floor(hiddenItems / 2);
+  let backHidden = Math.ceil(hiddenItems / 2);
+
+  if (itemsThatFit > 4) {
+    if (page <= frontHidden) {
+      // if current page would be in the front overflow
+      let newFrontHidden = frontHidden - (frontHidden - page) - 1;
+      newFrontHidden = newFrontHidden >= 0 ? newFrontHidden : 0;
+      const delta = frontHidden - newFrontHidden;
+
+      frontHidden -= delta;
+      backHidden += delta;
+    } else if (page >= totalItems - backHidden - 1) {
+      // if the current page would be in the back overflow
+      let newBackHidden = totalItems - 2 - page;
+      newBackHidden = newBackHidden >= 0 ? newBackHidden : 0;
+      const delta = backHidden - newBackHidden;
+
+      frontHidden += delta;
+      backHidden -= delta;
+    }
+  } else {
+    frontHidden = page > 1 ? page : 0;
+    frontHidden = frontHidden <= totalItems - 3 ? frontHidden : totalItems - 3;
+
+    backHidden = page < totalItems - 2 ? totalItems - page - 2 : 0;
+    backHidden = backHidden <= totalItems - 3 ? backHidden : totalItems - 3;
+  }
+
+  return {
+    front: frontHidden,
+    back: backHidden,
+  };
+}
+
+function DirectionButton({ direction, label, disabled, onClick }) {
+  const icon = direction === 'forward' ? CaretRight16 : CaretLeft16;
+
+  return (
+    <li className={`${prefix}--pagination-nav__list-item`}>
+      <Button
+        disabled={disabled}
+        renderIcon={icon}
+        kind="ghost"
+        hasIconOnly
+        iconDescription={label}
+        tooltipAlignment="center"
+        tooltipPosition="bottom"
+        onClick={onClick}
+      />
+    </li>
+  );
+}
+
+function PaginationItem({
+  page,
+  isActive,
+  onClick,
+  translateWithId: t = translateWithId,
+}) {
+  return (
+    <li className={`${prefix}--pagination-nav__list-item`}>
+      <button
+        className={classnames(`${prefix}--pagination-nav__page`, {
+          [`${prefix}--pagination-nav__page--active`]: isActive,
+        })}
+        onClick={onClick}>
+        <span className={`${prefix}--pagination-nav__accessibility-label`}>
+          {t('carbon.pagination-nav.item')}
+        </span>
+        {page}
+      </button>
+    </li>
+  );
+}
+
+function PaginationOverflow({
+  fromIndex,
+  count,
+  onSelect,
+  translateWithId: t = translateWithId,
+}) {
+  if (count > 1) {
+    return (
+      <li className={`${prefix}--pagination-nav__list-item`}>
+        <div className={`${prefix}--pagination-nav__select`}>
+          {/* eslint-disable-next-line jsx-a11y/no-onchange */}
+          <select
+            className={`${prefix}--pagination-nav__page ${prefix}--pagination-nav__page--select`}
+            aria-label={`Select ${t('carbon.pagination-nav.item')} number`}
+            onChange={e => {
```

Comment:
Should we be using `onBlur` here per the `jsx-a11y` rule?

Reply:
The issue with `onBlur` is that it only triggers once the user clicks on another element or continues with <kbd>tab</kbd>

![Screen Recording 2020-06-19 at 17 41 58](https://user-images.githubusercontent.com/28265588/85151778-7cdbd480-b254-11ea-8e71-eb46e3895771.gif)